File: minimizeAppToSystemTrayVB/Authentication.vb
Imports System.Windows.Forms

Public Class Authentication

    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        TextBox1.PasswordChar = "*"
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        If TextBox1.Text = "1234" Then
            RestoreMenu.Enabled = False
            HideMenu.Enabled = True
            Close()
            ShowWindow(WinDesktop, SW_SHOW)
        Else
            MessageBox.Show("Invalid authentication")
        End If

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Close()
    End Sub
End Class

File: minimizeAppToSystemTrayCSharp/Authentication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace minimizeAppToSystemTrayCSharp
{
    public partial class Authentication : Form
    {
        public Authentication()
        {
            InitializeComponent();
            AuthenticationTextBox.PasswordChar = '*';
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (Equals(AuthenticationTextBox.Text, "1234"))
            {
                Program.RestoreMenu.Enabled = false;
                Program.HideMenu.Enabled = true;
                Close();
                Program.ShowWindow(Program.WinConsole, Program.SW_SHOW);
            }
            else
            {
                MessageBox.Show("Invalid authentication");
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
